src/crimson/osd/scrub/scrub_validator.cc

```diff
@@ -60,9 +60,10 @@ shard_evaluation_t evaluate_object_shard(
   if (from == policy.primary) {
     ret.shard_info.primary = true;
   }
-  if (!maybe_obj || maybe_obj->negative) {
-    // impossible since chunky scrub was introduced
-    ceph_assert(!maybe_obj->negative);
+
+  // impossible since chunky scrub was introduced
+  ceph_assert(!maybe_obj->negative);
```

Comment:
maybe_obj may be nullptr

Reply:
ah yes, Assert should come after the check for nullptr.